frontend/benefit/handler/src/components/applicationReview/calculationTable/useCalculationTable.ts

```diff
@@ -0,0 +1,107 @@
+import { Calculation, Row } from 'benefit-shared/types/application';
+import clone from 'lodash/clone';
+import { convertToUIDateFormat, diffMonths } from 'shared/utils/date.utils';
+import { formatFloatToCurrency } from 'shared/utils/string.utils';
+
+export type BenefitRow = {
+  id: string;
+  dates: string;
+  amount: string;
+  amountNumber: string;
+  perMonth: string;
+  duration: number;
+  startDate: string;
+  endDate: string;
+};
+
+type Props = {
+  calculation: Calculation;
+};
+
+type CalculationTableProps = {
+  tableRows: Array<BenefitRow>;
+  totalSum: number;
+};
+
+const createBenefitRow = (): BenefitRow =>
+  clone({
+    id: '',
+    dates: '',
+    amount: '',
+    amountNumber: '',
+    perMonth: '',
+    duration: 0,
+    startDate: '',
+    endDate: '',
+  });
+
+const useCalculationTable = ({ calculation }: Props): CalculationTableProps => {
+  const filteredData: Row[] = calculation.rows.filter((row) =>
+    ['helsinki_benefit_monthly_eur', 'helsinki_benefit_sub_total_eur'].includes(
+      row.rowType
+    )
+  );
+
+  let tableRow = createBenefitRow();
+
+  const tableRows = filteredData.reduce((acc: BenefitRow[], row: Row) => {
+    tableRow.id = row.id;
+    if (row.rowType === 'helsinki_benefit_monthly_eur') {
+      tableRow.perMonth = `${formatFloatToCurrency(
+        row.amount,
+        'EUR',
+        'fi-FI',
+        0
+      )} / kk`;
+    }
+    if (row.rowType === 'helsinki_benefit_sub_total_eur') {
+      tableRow.dates = `${convertToUIDateFormat(
+        row.startDate
+      )} - ${convertToUIDateFormat(row.endDate)}`;
+
+      tableRow.endDate = convertToUIDateFormat(row.endDate);
+      tableRow.startDate = convertToUIDateFormat(row.startDate);
+
+      tableRow.duration = diffMonths(
+        new Date(row.endDate),
+        new Date(row.startDate)
+      );
+
+      tableRow.amount = formatFloatToCurrency(row.amount, 'EUR', 'fi-FI', 0);
+      tableRow.amountNumber = row.amount;
+      acc.push(tableRow);
+      tableRow = createBenefitRow();
+    }
+    return acc;
+  }, []);
+
+  const totalSum = tableRows.reduce(
+    (acc: number, cur: BenefitRow) => acc + parseFloat(cur.amountNumber),
+    0
+  );
+
+  if (tableRows.length > 0)
+    tableRows.push({
+      ...createBenefitRow(),
+      id: 'table-footer',
+      dates: `${tableRows.at(0).startDate} - ${tableRows.at(-1).endDate}`,
+      // Summing up multiple duration periods may cause float number inaccuracies to accumulate;
```

Comment:
Note: while most of this file is moved as-is from CalculationReview, this formatting part here is new in this PR.